taf/validation.py

```diff
@@ -90,8 +91,14 @@ def validate_branch(
                     unmodified_roles_and_versions[role_name] = version
 
             if updated_role in check_branch_roles:
+                no_initial_branch_id = check_branch_roles[updated_role]
                 branch_id = _check_branch_id(
-                    auth_repo, auth_commit, branch_id, updated_role
+                    auth_repo,
+                    auth_commit,
+                    branch_id,
+                    updated_role,
+                    is_first_commit=no_initial_branch_id
+                    and commit_index == len(auth_commits) - 1,
```

Comment:
Approved. Minor comment, non blocking: would it maybe be simpler if we skip calling `_check_branch_id`  in case of `is_first_commit` being True?

Reply:
That makes sense, but what we really want to do is implement this more generically. I dislike this solution, this is the part of the code that I kept implementing and losing due to hard resets and I would like to come up with a better way of handling this